### ISSUE TITLE
Godot shader matching bracket checker does not ignore brackets in comments

### ISSUE THREAD

**Issue by AntonBogun** [godotengine/godot#53332]
### Godot version

v3.3.3.stable.mono.official.b973f997f

### System information

Windows 10, GLES3, GeForce RTX 3050 Ti Laptop GPU/PCIe/SSE2

### Issue description

Shader bracket checker still runs the check on brackets in comments, which can make developing difficult
![image](https://user-images.githubusercontent.com/70374466/135713489-2bdf16e1-25e5-49da-be8b-64c81689d21d.png)
![image](https://user-images.githubusercontent.com/70374466/135713492-60df52a4-6823-466f-9b66-036d2233439b.png)

### Steps to reproduce

Put unmatching amount of brackets in a comment between matching brackets

### Minimal reproduction project

[Shader Brackets.zip](https://github.com/godotengine/godot/files/7271328/Shader.Brackets.zip)

**Comment by Calinou** [comment]
Does this happen in GDScript too?

**Comment by AntonBogun** [comment]
Surprisingly, yes, after more testing this appears to be an issue present in any editing interface inside the editor

**Comment by ChrisKonishi** [comment]
I'm new to the Godot Project and I'd like to start contributing. I've looked at the Junior Jobs, but they seem to be all taken at the moment. Could this be an issue that I can take?

**Comment by akien-mga** [comment]
Yes I think that's a good one to work on for a new contributor. The issue seems reproducible in the `master` branch too, the relevant code might be in `editor/script_text_editor.cpp` (or possibly in `scene/gui/code_edit.cpp` which is used in the ScriptTextEditor.

**Comment by ChrisKonishi** [comment]
Thanks! I'll look into it

**Comment by ChrisKonishi** [comment]
Thanks, I'll look into it!

**Comment by Cevantime** [comment]
As a junior too, I think I fixed this, but only for 3.x ! 
So @ChrisKonishi  you still have a chance to prove your worth on the master branch :)

**Comment by ChrisKonishi** [comment]
@Cevantime, it seems to be an elegant solution, I've been working on another idea, just using an existing function to ignore commented brackets, maybe not so appropriate. Congrats!

**Comment by gustavi** [comment]
All "closing something" or "opening something" characters are affected.

```gdscript
func opening():
    hide(
        # (
    )
    var dict = {
        # {
    }
    var array = [
        # [
    ]

func closing():
    hide(
        # )
    )
    var dict = {
        # }
    }
    var array = [
        # ]
    ]
```

Tested on Godot 3.3, 3.4 and 4.0 (21b81998bb42b2324a539f01354b68e4fba2c696).
